src/common/viz/FigureExtractor.js

```diff
@@ -0,0 +1,145 @@
+/*globals define, _*/
+define(['./Utils'], function (Utils) {
+    const FigureExtractor = function (client) {
+        this._client = client;
+        this._metaNodesMap = this._initializeMetaNodesMap();
+    };
+    const EXTRACTORS = {
+        GRAPH: 'Graph',
+        SUB_GRAPH: 'SubGraph',
+        IMAGE: 'Image',
+        LINE: 'Line',
+    };
+
+    FigureExtractor.prototype._initializeMetaNodesMap = function () {
+        const metaNodes = this._client.getAllMetaNodes();
+        const idsAndTypes = metaNodes.map(node => [node.getId(), node.getAttribute('name')]);
+        return _.object(idsAndTypes);
+    };
+
+    FigureExtractor.prototype.extract = function(node) {
+        const extractorFn = this.getMetaType(node);
+        return this[extractorFn](node);
+    };
+
+    FigureExtractor.prototype.constructor = FigureExtractor;
+
+    FigureExtractor.prototype[EXTRACTORS.GRAPH] = function(node) {
+        const [id, jobId, execId] = this.getIdHierarchy(node, 3); // eslint-disable-line no-unused-vars
```

Comment:
Rather than disable the eslint rule, I would either omit `jobId` or comment it out:
```javascript
const [id, /*jobId*/, execId] = this.getIdHierarchy(node, 3);
```